### PR TITLE
tests: randomizer should always return unique message

### PR DESCRIPTION
**Pull request by batunpc**
As discussed in https://github.com/pavelkeyzik/pr-approve-generator/issues/2 I have integrated the `istanbul` coverage provider and set it up as script `npm run coverage` in `package.json`. This command will provide `index.html` file within the directory `coverage`, as best practices I added this into `.gitignore`. 

Currently I am covering only 83% of the `randomizer.js` file. I will turn this into draft PR and let you know when its covered 100% 🚀

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,10 +6,11 @@
   "scripts": {
     "dev": "vite",
     "build": "vite build",
-    "preview": "vite preview",
```
Comment (pavelkeyzik) [inline package.json:9]: May ask you why `preview` script was removed? 🤔

Comment (batunpc) [reply package.json:9]: My apologies, I think I have overwritten somehow while adding scripts for test

Comment (batunpc) [reply package.json:9]: Fixing asap :)

File: src/core/randomizer.js [hunk 2]
```diff
@@ -15,13 +15,13 @@ function buildRandomizer() {
       let newMessagesState;
 
       if (messagesState.length !== 0) {
-        newMessage = messagesState.at(index);
```
Comment (pavelkeyzik) [inline src/core/randomizer.js:18]: Was this replaces automatically by IDE? Is there any concerns using `.at(index)` over `[index]`. I'm okay with `[index]` too, just wanted to know you opinion 😊

File: src/core/tests/randomizer.test.js [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
-// The logic here is that for example we have 5 messages, we generated one
-// and removed from messages array, until our array will be empty.
-// When it's empty we just replaces this array with the initial array
 it("should always return unique message", () => {
-  expect(true).toBe(true);
+  const randomizer = buildRandomizer();
+  const messages = Array.from({ length: 5 }, () =>
+    randomizer.getRandomMessage()
+  );
+  const uniqueMessages = new Set(messages);
```
Comment (pavelkeyzik) [inline src/core/tests/randomizer.test.js:9]: I love this idea 👍

File: package.json (hunk 1 above)
Comment (pavelkeyzik) [reply package.json:9]: No rush 👍 Thanks for working on that 😊

File: src/core/randomizer.js (hunk 2 above)
Comment (batunpc) [reply src/core/randomizer.js:18]: No not IDE , it was intentional, I thought it could be more readable this way 😅

**Comment by batunpc** [comment]
Screenshot to present coverage for the record

CleanShot 2022-10-16 at 05 21 08@2x

File: src/core/randomizer.js (hunk 2 above)
Comment (pavelkeyzik) [reply src/core/randomizer.js:18]: Fine! Let's keep it then 😆

**Comment by batunpc** [comment]
Hey @pavelkeyzik, it's all done! Let me know if I should improve anything else 🤩

**Comment by pavelkeyzik** [comment]
Actually... Yes! I have one suggestion. As you may see in `randomizer` file, we just import these messages from file. It'll be better if we provide this as a parameter `buildRandomizer(messages)`. That way we could mock our data in unit tests and check if result is correct.

At the moment, we have `should always return unique message` test and if we'll change length 5 to 50 it will fail, because we have only 20 unique messages at the moment. Instead, we can provide mocked array with 5 elements and generate 5 times and make sure that all messages unique. I know it's a tiny change, but I think it'll make more sense. Any thoughts? 🤔 

The second test `should return initial array when the array is empty` works pretty much same way as the first one. I'd suggest to update `getRandomMessage` function to return `currentMessage` and `messageState`, to be able check `messageState` in unit tests. We can provide 5 messages, generate 4 times, check if it's only one element, generate one more time and check that `messageState` contains 5 messages again. Please, ask more questions about this case, as sometimes my explanation not so good 😆

**Comment by batunpc** [comment]
>  Instead, we can provide mocked array with 5 elements and generate 5 times and make sure that all messages unique.

I believe I understood, I will work on this as soon as I get chance!👍  Thanks for your insights

**Comment by batunpc** [comment]
Hey @pavelkeyzik, I am working on your comments
> We can provide 5 messages, generate 4 times, check if it's only one element, generate one more time and check that messageState contains 5 messages again

During this draft PR I will submit something and we can work from there, when I am satisfied with my changes I will mark this as ready for review ☺️

**Comment by batunpc** [comment]
Hey @pavelkeyzik! couple of updates ✨

1. I have updated the `buildRandomizer(messages)` along with its tests. Please let me know your thoughts. 🤩
2. Additionally I would like to also suggest another devDependecy which is linter (i.e. [eslint](https://www.npmjs.com/package/eslint)) to follow patterns of `ECMAScript/JavaScript code`. This idea came to my mind from my other JS projects where I used it a lot. We could [configure Eslint](https://eslint.org/docs/latest/user-guide/configuring/) with the file `.eslintrc.js` and set some conditions as desired.

**Comment by batunpc** [comment]
If eslint sounds good to you I would turn it into draft and integrate it as well, let me know :-)

File: src/core/randomizer.js [hunk 4]
```diff
@@ -1,37 +1,31 @@
-import { messages } from "./messages";
-
-function buildRandomizer() {
+function buildRandomizer(messages) {
   const state = {
-    currentMessage: messages[0],
+    messages: [...messages],
```
Comment (pavelkeyzik) [inline src/core/randomizer.js:3]: I think we can just do something like this:

```suggestion
    messages,
```

File: src/core/tests/randomizer.test.js [hunk 5]
```diff
@@ -1,11 +1,21 @@
...
+
+it("should return initial array when the array is empty", () => {
+  const messages = ["1", "2", "3", "4", "5"];
+  const randomizer = buildRandomizer(messages);
+  for (let i = 0; i < 6; i++) {
+    const { newMessage, newMessagesState } = randomizer.getRandomMessage();
+    expect(newMessagesState).not.toContain(newMessage);
+  }
+  const { newMessage, newMessagesState } = randomizer.getRandomMessage();
+  expect(newMessagesState).not.toContain(newMessage);
```
Comment (pavelkeyzik) [inline src/core/tests/randomizer.test.js:20]: When I was talking about checking if newMessagesState contain new array, I meant something like this:

```js
for (let i = 0; i < message.length; i++) {
    const { newMessagesState } = randomizer.getRandomMessage();
    
    if (i === messages.length - 1) {
      expect(newMessagesState).toHaveLength(messages.length - 1);
    }
  }
```

**Comment by pavelkeyzik** [comment]
@batunpc It'll be nice to add ESLint 😊 Let's do this as part of the separate ticket #6

File: src/core/tests/randomizer.test.js (hunk 5 above)
Comment (batunpc) [reply src/core/tests/randomizer.test.js:20]: Ah I see

File: src/core/randomizer.js (hunk 4 above)
Comment (batunpc) [reply src/core/randomizer.js:3]: Alright 👍

File: src/core/tests/randomizer.test.js (hunk 5 above)
Comment (batunpc) [reply src/core/tests/randomizer.test.js]: Done! 👍

File: src/core/tests/randomizer.test.js [hunk 6]
```diff
@@ -1,11 +1,20 @@
...
+});
+
+it("should return initial array when the array is empty", () => {
+  const messages = ["1", "2", "3", "4", "5"];
+  const randomizer = buildRandomizer(messages);
+
+  for (let i = 0; i < messages.length; i++) {
+    const { newMessagesState } = randomizer.getRandomMessage();
+    if (i === messages.length - i) expect(newMessagesState).toEqual(messages);
```
[the remark this answers is not in the document]
Comment (batunpc) [reply src/core/tests/randomizer.test.js:18]: Hi @pavelkeyzik of course, I will be updating this now! Thanks lot for your comments, and happy Hacktoberfest! 🎃

**Review by pavelkeyzik — APPROVED** [review]
Thanks for working on this ticket. Happy Hacktoberfest to you too 🎃

**Pull request merged**